numpy/version.pyi

```diff
@@ -1,7 +1,24 @@
-version: str
-__version__: str
-full_version: str
+import sys
+from typing import Final, TypeAlias
 
-git_revision: str
-release: bool
-short_version: str
+if sys.version_info >= (3, 11):
+    from typing import LiteralString
+else:
+    LiteralString: TypeAlias = str
+
+__all__ = (
+    '__version__',
+    'full_version',
+    'git_revision',
+    'release',
+    'short_version',
+    'version',
+)
```

Comment:
Curious, having `__all__` is important for types in some way?

Reply:
In this case it should make a functional difference. 
I mostly added it because "explicit is better than implicit", and because makes it easier for typecheckers to determine the public module members, which can help with performance.